lib/mqtt/MqttController.js

```diff
@@ -0,0 +1,248 @@
+const RobotMqttHandle = require("./handles/RobotMqttHandle");
+const MqttCommonAttributes = require("./MqttCommonAttributes");
+const HomieCommonAttributes = require("./homie/HomieCommonAttributes");
+const mqtt = require("mqtt");
+const asyncMqtt = require("async-mqtt");
+const Logger = require("../Logger");
+
+class MqttController {
+    /**
+     * @param {object} options
+     * @param {import("../core/ValetudoRobot")} options.robot
+     * @param {import("../Configuration")} options.config
+     */
+    constructor(options) {
+        this.loadConfig();
+
+        this.client = null;
+        // Workaround to be able to set two Last Will and Testament payloads, one for Homie and one for HAss
+        // this.clientSecondaryLWT = null; // TODO actually implement
+
+        /** @type {import(./NodeMqttHandler)} */
+        this.nodes = [];
+
+        /**
+         * @callback subscribeCb
+         * @param {string} value
+         * @return {Promise<void>}
+         */
+        /** @type {object.<subscribeCb>} */
+        this.subscriptions = {};
+
+        this.initialized = false;
+
+        this.loadConfig();
+
+        this.robotHandle = new RobotMqttHandle({
+            robot: options.robot,
+            controller: this,
+            baseTopic: this.topicPrefix,
+            topicName: this.identifier,
+            friendlyName: this.friendlyName
+        });
+
+        if (this.enabled) {
+            this.connect().then();
```

Comment:
Retry logic?

Reply:
Yes, definitely planned

Reply:
It turns out that MQTT.js handles this internally. Both cases "broker unavailable on start" and "broker becomes unavailable" are handled, it will just spam the log once per second but I think it's okay.

```
[2021-04-05T01:16:44.496Z] [ERROR] MQTT error: Error: connect ECONNREFUSED 10.69.0.13:1883
[2021-04-05T01:16:45.499Z] [ERROR] MQTT error: Error: connect ECONNREFUSED 10.69.0.13:1883
[2021-04-05T01:16:46.499Z] [ERROR] MQTT error: Error: connect ECONNREFUSED 10.69.0.13:1883
[2021-04-05T01:16:47.502Z] [ERROR] MQTT error: Error: connect ECONNREFUSED 10.69.0.13:1883
[2021-04-05T01:16:48.504Z] [ERROR] MQTT error: Error: connect ECONNREFUSED 10.69.0.13:1883
```